### PR TITLE
Refactor setup.py for cmake updates

### LINKED ISSUE #666: Improve unboxing experience for users without cmake installed
Right now, if a user does not have cmake installed and they try to install, they receive a very verbose message (see example at end of ticket).

Ideally, users would:
1. receive a very short and friendly message telling them they should install cmake
2. have wheels provided for the majority of common platforms to make sure they don't encounter this

Question: Can cmake produce something like a Makefile that we include to avoid having a dependency on cmake for source installs?

example message:

```
Installing collected packages: opentimelineio
  Running setup.py install for opentimelineio ... error
    ERROR: Command errored out with exit status 1:
     command: /Users/username/dev/otiowheel/venv-otiowheel/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/gw/8vl6g7s91dq3sshvy3hqtftc0000gn/T/pip-install-di_hdnwz/opentimelineio/setup.py'"'"'; __file__='"'"'/private/var/folders/gw/8vl6g7s91dq3sshvy3hqtftc0000gn/T/pip-install-di_hdnwz/opentimelineio/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/gw/8vl6g7s91dq3sshvy3hqtftc0000gn/T/pip-record-889gs1ef/install-record.txt --single-version-externally-managed --compile --install-headers /Users/username/dev/otiowheel/venv-otiowheel/include/site/python3.8/opentimelineio
         cwd: /private/var/folders/gw/8vl6g7s91dq3sshvy3hqtftc0000gn/T/pip-install-di_hdnwz/opentimelineio/
    Complete output (169 lines):
    running install
    running build
    running build_py
    creating build
    creating build/lib.macosx-10.9-x86_64-3.8
    creating build/lib.macosx-10.9-x86_64-3.8/opentimelineio
    copying src/py-opentimelineio/opentimelineio/test_utils.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio
    copying src/py-opentimelineio/opentimelineio/opentime.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio
    copying src/py-opentimelineio/opentimelineio/hooks.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio
    copying src/py-opentimelineio/opentimelineio/__init__.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio
    copying src/py-opentimelineio/opentimelineio/media_linker.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio
    copying src/py-opentimelineio/opentimelineio/exceptions.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio
    creating build/lib.macosx-10.9-x86_64-3.8/opentimelineio/core
    copying src/py-opentimelineio/opentimelineio/core/_core_utils.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/core
    copying src/py-opentimelineio/opentimelineio/core/__init__.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/core
    copying src/py-opentimelineio/opentimelineio/core/mediaReference.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/core
    copying src/py-opentimelineio/opentimelineio/core/composable.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/core
    copying src/py-opentimelineio/opentimelineio/core/composition.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/core
    copying src/py-opentimelineio/opentimelineio/core/item.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/core
    creating build/lib.macosx-10.9-x86_64-3.8/opentimelineio/plugins
    copying src/py-opentimelineio/opentimelineio/plugins/manifest.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/plugins
    copying src/py-opentimelineio/opentimelineio/plugins/__init__.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/plugins
    copying src/py-opentimelineio/opentimelineio/plugins/python_plugin.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/plugins
    creating build/lib.macosx-10.9-x86_64-3.8/opentimelineio/algorithms
    copying src/py-opentimelineio/opentimelineio/algorithms/track_algo.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/algorithms
    copying src/py-opentimelineio/opentimelineio/algorithms/stack_algo.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/algorithms
    copying src/py-opentimelineio/opentimelineio/algorithms/timeline_algo.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/algorithms
    copying src/py-opentimelineio/opentimelineio/algorithms/__init__.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/algorithms
    copying src/py-opentimelineio/opentimelineio/algorithms/filter.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/algorithms
    creating build/lib.macosx-10.9-x86_64-3.8/opentimelineio/adapters
    copying src/py-opentimelineio/opentimelineio/adapters/otio_json.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/adapters
    copying src/py-opentimelineio/opentimelineio/adapters/adapter.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/adapters
    copying src/py-opentimelineio/opentimelineio/adapters/__init__.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/adapters
    copying src/py-opentimelineio/opentimelineio/adapters/cmx_3600.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/adapters
    copying src/py-opentimelineio/opentimelineio/adapters/fcp_xml.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/adapters
    creating build/lib.macosx-10.9-x86_64-3.8/opentimelineio/schema
    copying src/py-opentimelineio/opentimelineio/schema/timeline.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/schema
    copying src/py-opentimelineio/opentimelineio/schema/track.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/schema
    copying src/py-opentimelineio/opentimelineio/schema/transition.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/schema
    copying src/py-opentimelineio/opentimelineio/schema/external_reference.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/schema
    copying src/py-opentimelineio/opentimelineio/schema/__init__.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/schema
    copying src/py-opentimelineio/opentimelineio/schema/serializable_collection.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/schema
    copying src/py-opentimelineio/opentimelineio/schema/schemadef.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/schema
    copying src/py-opentimelineio/opentimelineio/schema/marker.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/schema
    copying src/py-opentimelineio/opentimelineio/schema/stack.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/schema
    copying src/py-opentimelineio/opentimelineio/schema/clip.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/schema
    copying src/py-opentimelineio/opentimelineio/schema/generator_reference.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/schema
    copying src/py-opentimelineio/opentimelineio/schema/effect.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/schema
    copying src/py-opentimelineio/opentimelineio/schema/foo.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/schema
    creating build/lib.macosx-10.9-x86_64-3.8/opentimelineio/schemadef
    copying src/py-opentimelineio/opentimelineio/schemadef/__init__.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/schemadef
    creating build/lib.macosx-10.9-x86_64-3.8/opentimelineio/console
    copying src/py-opentimelineio/opentimelineio/console/otiostat.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/console
    copying src/py-opentimelineio/opentimelineio/console/otiocat.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/console
    copying src/py-opentimelineio/opentimelineio/console/otioconvert.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/console
    copying src/py-opentimelineio/opentimelineio/console/console_utils.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/console
    copying src/py-opentimelineio/opentimelineio/console/__init__.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/console
    copying src/py-opentimelineio/opentimelineio/console/otiopluginfo.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/console
    copying src/py-opentimelineio/opentimelineio/console/autogen_plugin_documentation.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/console
    copying src/py-opentimelineio/opentimelineio/console/autogen_serialized_datamodel.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/console
    creating build/lib.macosx-10.9-x86_64-3.8/opentimelineview
    copying src/opentimelineview/console.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineview
    copying src/opentimelineview/ruler_widget.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineview
    copying src/opentimelineview/timeline_widget.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineview
    copying src/opentimelineview/__init__.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineview
    copying src/opentimelineview/details_widget.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineview
    copying src/opentimelineview/settings.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineview
    copying src/opentimelineview/track_widgets.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineview
    creating build/lib.macosx-10.9-x86_64-3.8/opentimelineio_contrib
    copying contrib/opentimelineio_contrib/__init__.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio_contrib
    creating build/lib.macosx-10.9-x86_64-3.8/opentimelineio_contrib/adapters
    copying contrib/opentimelineio_contrib/adapters/kdenlive.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio_contrib/adapters
    copying contrib/opentimelineio_contrib/adapters/extern_rv.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio_contrib/adapters
    copying contrib/opentimelineio_contrib/adapters/advanced_authoring_format.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio_contrib/adapters
    copying contrib/opentimelineio_contrib/adapters/ffmpeg_burnins.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio_contrib/adapters
    copying contrib/opentimelineio_contrib/adapters/extern_maya_sequencer.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio_contrib/adapters
    copying contrib/opentimelineio_contrib/adapters/__init__.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio_contrib/adapters
    copying contrib/opentimelineio_contrib/adapters/fcpx_xml.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio_contrib/adapters
    copying contrib/opentimelineio_contrib/adapters/maya_sequencer.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio_contrib/adapters
    copying contrib/opentimelineio_contrib/adapters/rv.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio_contrib/adapters
    copying contrib/opentimelineio_contrib/adapters/xges.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio_contrib/adapters
    copying contrib/opentimelineio_contrib/adapters/ale.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio_contrib/adapters
    copying contrib/opentimelineio_contrib/adapters/hls_playlist.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio_contrib/adapters
    copying contrib/opentimelineio_contrib/adapters/burnins.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio_contrib/adapters
    creating build/lib.macosx-10.9-x86_64-3.8/opentimelineio_contrib/adapters/aaf_adapter
    copying contrib/opentimelineio_contrib/adapters/aaf_adapter/__init__.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio_contrib/adapters/aaf_adapter
    copying contrib/opentimelineio_contrib/adapters/aaf_adapter/aaf_writer.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio_contrib/adapters/aaf_adapter
    running egg_info
    writing OpenTimelineIO.egg-info/PKG-INFO
    writing dependency_links to OpenTimelineIO.egg-info/dependency_links.txt
    writing entry points to OpenTimelineIO.egg-info/entry_points.txt
    writing requirements to OpenTimelineIO.egg-info/requires.txt
    writing top-level names to OpenTimelineIO.egg-info/top_level.txt
    reading manifest file 'OpenTimelineIO.egg-info/SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    warning: no files found matching 'CHANGELOG.md'
    no previously-included directories found matching 'docs'
    warning: no previously-included files found matching '.gitmodules'
    warning: no previously-included files matching '*.git' found under directory 'src'
    warning: no previously-included files found matching '.readthedocs.yml'
    warning: no previously-included files found matching 'readthedocs-conda.yml'
    warning: no previously-included files found matching '.codecov.yml'
    warning: no previously-included files found matching '.gitlab-ci.yml'
    warning: no previously-included files found matching '.travis.yml'
    warning: no previously-included files found matching '*.pdf'
    warning: no previously-included files found matching 'CODE_OF_CONDUCT.md'
    warning: no previously-included files found matching 'CONTRIBUTING.md'
    warning: no previously-included files found matching 'contrib/opentimelineio_contrib/adapters/Makefile'
    warning: no previously-included files found matching 'Makefile'
    no previously-included directories found matching 'contrib/opentimelineio_contrib/adapters/tests'
    no previously-included directories found matching 'tests'
    no previously-included directories found matching 'src/deps/pybind11/tools/clang'
    no previously-included directories found matching 'src/deps/rapidjson/thirdparty'
    writing manifest file 'OpenTimelineIO.egg-info/SOURCES.txt'
    copying src/py-opentimelineio/opentimelineio/adapters/builtin_adapters.plugin_manifest.json -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio/adapters
    creating build/lib.macosx-10.9-x86_64-3.8/opentimelineio_contrib/application_plugins
    creating build/lib.macosx-10.9-x86_64-3.8/opentimelineio_contrib/application_plugins/rv
    creating build/lib.macosx-10.9-x86_64-3.8/opentimelineio_contrib/application_plugins/rv/example_otio_reader
    copying contrib/opentimelineio_contrib/application_plugins/rv/example_otio_reader/PACKAGE -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio_contrib/application_plugins/rv/example_otio_reader
    copying contrib/opentimelineio_contrib/application_plugins/rv/example_otio_reader/example_otio_reader_plugin.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio_contrib/application_plugins/rv/example_otio_reader
    copying contrib/opentimelineio_contrib/application_plugins/rv/example_otio_reader/otio_reader.py -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio_contrib/application_plugins/rv/example_otio_reader
    copying contrib/opentimelineio_contrib/adapters/contrib_adapters.plugin_manifest.json -> build/lib.macosx-10.9-x86_64-3.8/opentimelineio_contrib/adapters
    running build_ext
    Traceback (most recent call last):
      File "/private/var/folders/gw/8vl6g7s91dq3sshvy3hqtftc0000gn/T/pip-install-di_hdnwz/opentimelineio/setup.py", line 171, in run
        out = subprocess.check_output(['cmake', '--version'])
      File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/subprocess.py", line 411, in check_output
        return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
      File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/subprocess.py", line 489, in run
        with Popen(*popenargs, **kwargs) as process:
      File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/subprocess.py", line 854, in __init__
        self._execute_child(args, executable, preexec_fn, close_fds,
      File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/subprocess.py", line 1702, in _execute_child
        raise child_exception_type(errno_num, err_msg, err_filename)
    FileNotFoundError: [Errno 2] No such file or directory: 'cmake'

    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/var/folders/gw/8vl6g7s91dq3sshvy3hqtftc0000gn/T/pip-install-di_hdnwz/opentimelineio/setup.py", line 358, in <module>
        setup(
      File "/Users/username/dev/otiowheel/venv-otiowheel/lib/python3.8/site-packages/setuptools/__init__.py", line 145, in setup
        return distutils.core.setup(**attrs)
      File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/distutils/core.py", line 148, in setup
        dist.run_commands()
      File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/distutils/dist.py", line 966, in run_commands
        self.run_command(cmd)
      File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/distutils/dist.py", line 985, in run_command
        cmd_obj.run()
      File "/private/var/folders/gw/8vl6g7s91dq3sshvy3hqtftc0000gn/T/pip-install-di_hdnwz/opentimelineio/setup.py", line 140, in run
        install.run(self)
      File "/Users/username/dev/otiowheel/venv-otiowheel/lib/python3.8/site-packages/setuptools/command/install.py", line 61, in run
        return orig.install.run(self)
      File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/distutils/command/install.py", line 545, in run
        self.run_command('build')
      File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/distutils/cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/distutils/dist.py", line 985, in run_command
        cmd_obj.run()
      File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/distutils/command/build.py", line 135, in run
        self.run_command(cmd_name)
      File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/distutils/cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/distutils/dist.py", line 985, in run_command
        cmd_obj.run()
      File "/private/var/folders/gw/8vl6g7s91dq3sshvy3hqtftc0000gn/T/pip-install-di_hdnwz/opentimelineio/setup.py", line 173, in run
        raise RuntimeError(
    RuntimeError: CMake must be installed to build the following extensions: _opentimelineio, _opentime
    ----------------------------------------
ERROR: Command errored out with exit status 1: /Users/username/dev/otiowheel/venv-otiowheel/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/gw/8vl6g7s91dq3sshvy3hqtftc0000gn/T/pip-install-di_hdnwz/opentimelineio/setup.py'"'"'; __file__='"'"'/private/var/folders/gw/8vl6g7s91dq3sshvy3hqtftc0000gn/T/pip-install-di_hdnwz/opentimelineio/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/gw/8vl6g7s91dq3sshvy3hqtftc0000gn/T/pip-record-889gs1ef/install-record.txt --single-version-externally-managed --compile --install-headers /Users/username/dev/otiowheel/venv-otiowheel/include/site/python3.8/opentimelineio Check the logs for full command output.
```

### PR DESCRIPTION
**Pull request by reinecke**
Refactors `setup.py` to properly leverage the newly cleaned up cmake builds (reworked in #837).

**Summarize your change.**

Our `setup.py` had been arranged to try and figure out the final install location for for the libraries and place products in there. The python packaging utilities actually want you to assemble your build into a temp location where it will then pick it up to either package or install to the proper place.

This refactor now tells the cmake build to install into that temp location. Because of this, we have to manually maintain much less code that supports determining the user's desired end action with the assembled package (e.x. install, install using --user, bdist, bdist_wheel).

Also removed `_Ctx` class instance that was storing build config as a global - it made it hard to follow the flow of code and would likely be prone to bugs.

### CODE REVIEW

**Comment by reinecke** [comment]
For windows, I'm pretty certain we need to at least pull:

https://github.com/pybind/cmake_example/blob/6de957d06a372a7b106e5e3b8e29b97a8ec3afc7/setup.py#L10-L15

and:

https://github.com/pybind/cmake_example/blob/6de957d06a372a7b106e5e3b8e29b97a8ec3afc7/setup.py#L73

And replace this:

https://github.com/PixarAnimationStudios/OpenTimelineIO/blob/db9996779cc80bf4268d0c3b2f1f8f80cfd55154/setup.py#L129-L130

@meshula notes:
> also see this confusing stuff: https://github.com/PixarAnimationStudios/USD/blob/090ef0d849ced875b3b99a15e454b69148ccd8e1/build_scripts/build_usd.py#L337

> Notice that VS2019 and greater has an inconvenient new incompatible syntax.
> There's a wodge of code here that detects the species of Visual Studio in the active build environment: https://github.com/PixarAnimationStudios/USD/blob/090ef0d849ced875b3b99a15e454b69148ccd8e1/build_scripts/build_usd.py#L133

**Comment by ssteinbach** [comment]
this is a pretty substantial change to the setup.py (but one that looks good!). could we break the cmake setup.py stuff out away from the cibuildwheel stuff?

**Review by ssteinbach — CHANGES_REQUESTED** [review]
I think I'd prefer if we could break this into two PRs - one for the setup.py changes and one for the python-wheels/cibuildwheel action addition that follows it.

**Comment by reinecke** [comment]
Sure, I could make the first one be "properly target build output from setup.py" and then make the second one be adding the action and publishing the wheel artifacts.

**Comment by reinecke** [comment]
Oh, it also looks like I may have branched from an old commit. I'll rebase and split.

**Comment by ssteinbach** [comment]
Great, thanks @reinecke.

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[build-system]
+requires = [
+    "setuptools>=20.5.0",
+    "wheel",
+    "cmake>=3.12",
+    "pip>=6.0.0",
```
[the remark this answers is not in the document]
Comment (meshula) [reply pyproject.toml:6]: cmake doesn't need it.

File: setup.py [hunk 2]
```diff
@@ -87,141 +64,105 @@ def cmake_version_check():
...
 
     def run(self):
-        _ctx.cxx_coverage = (
-            self.cxx_coverage is not False
-            or bool(os.environ.get("OTIO_CXX_COVERAGE_BUILD"))
-        )
-
+        # This works around the fact that we build _opentime and _otio
+        # extensions as a one-shot cmake invocation. Usually we'd build each
+        # separate using build_extension.
```
Comment (meshula) [inline setup.py:86]: typo separately

File: setup.py [hunk 3]
```diff
@@ -87,141 +64,105 @@ def cmake_version_check():
...
+        )
+
+        cmake_args += [
+            '-DOTIO_PYTHON_INSTALL_DIR=' + install_dir,
+            '-DCMAKE_INSTALL_PREFIX=' + cmake_install_prefix,
+        ]
+
+        if platform.system() == "Windows":
+            if sys.maxsize > 2**32:
+                cmake_args += ['-A', 'x64']
```
Comment (meshula) [inline setup.py:132]: per usd_build.py, we should detect VS2019 or greater, as this argument doesn't apply for VS2017 and less, where 64 bit builds are indicated by appending "Win64" to the build target.

File: setup.py [hunk 4]
```diff
@@ -87,141 +64,105 @@ def cmake_version_check():
...
+            cmake_args,
+            cwd=self.build_temp_dir,
+            env=os.environ.copy()
+        )
+
+    def cmake_install(self):
+        if platform.system() == "Windows":
+            multi_proc = '/m'
+        else:
+            multi_proc = '-j2'
```
Comment (meshula) [inline setup.py:156]: conservative :)

**Review by meshula — COMMENTED** [review]
Great to see! Very happy to have wheels built as an action.

File: setup.py (hunk 4 above)
Comment (reinecke) [reply setup.py:156]: Hah, I just relocated that line from the old setup. Probably worth re-evaluating at that :)

File: pyproject.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ssteinbach) [reply pyproject.toml:6]: We originally needed to check setup-tools and pip versions because on _very_ ancient distributions, neither would actually do version checking for itself (meaning pip/setup-tools wouldn't detect if the version of setup-tools needed to be updated).  At this point I think we can drop the local requirement since those are prerequisites of installing via pip/setup.py itself... unless we're using features present in those libraries after that point.  At least at one point we were using stuff from "more modern" setup-tools.

File: setup.py (hunk 4 above)
Comment (ssteinbach) [reply setup.py:156]: if we want to detect the number of cores:

`multiprocessing.cpu_count()`

File: setup.py (hunk 3 above)
Comment (reinecke) [reply setup.py:132]: Moved this conversation over to the new PR: https://github.com/PixarAnimationStudios/OpenTimelineIO/pull/956/files#r622417623
